services/venice-controller/src/main/java/com/linkedin/venice/controller/server/ControllerRoutes.java

```diff
@@ -179,6 +187,42 @@ private Route updateKafkaTopicConfig(Admin admin, UpdateTopicConfigFunction upda
     };
   }
 
+  public Route getAggregatedHealthStatus(Admin admin) {
+    return (request, response) -> {
+      StoppableNodeStatusResponse responseObject = new StoppableNodeStatusResponse();
+      response.type(HttpConstants.JSON);
+
+      try {
+        AggregratedHealthStatusRequest statusRequest =
+            OBJECT_MAPPER.readValue(request.body(), AggregratedHealthStatusRequest.class);
+        String cluster = statusRequest.getClusterId();
+        List<String> instanceList = statusRequest.getInstances();
+        if (instanceList.isEmpty()) {
+          responseObject.setError("Empty instances list");
+          responseObject.setErrorType(ErrorType.BAD_REQUEST);
+          response.status(HttpStatus.SC_BAD_REQUEST);
+          return AdminSparkServer.OBJECT_MAPPER.writeValueAsString(responseObject);
+        }
+
+        List<String> toBeStoppedInstanceList = statusRequest.getToBeStoppedInstances();
+        responseObject.setCluster(cluster);
+
+        InstanceRemovableStatuses statuses =
+            admin.getAggregatedHealthStatus(cluster, instanceList, toBeStoppedInstanceList);
+        if (statuses.getRedirectUrl() != null) {
+          response.redirect(statuses.getRedirectUrl() + AGGR_HEALTH_STATUS_URI, HttpStatus.SC_MOVED_TEMPORARILY);
```

Comment:
I still think we should not return a response in case we're returning a `SC_MOVED_TEMPORARILY` status code